fmriprep/interfaces/fmap.py

```diff
@@ -530,3 +584,30 @@ def _delta_te(in_values, te1=None, te2=None):
             'EchoTime2 metadata field not found. Please consult the BIDS specification.')
 
     return abs(float(te2) - float(te1))
+
+def _delta_te_from_two_phases(in_dicts, te1=None, te2=None):
+    """Read :math:`\Delta_\text{TE}` from two BIDS metadata dicts"""
+
+    if isinstance(in_values, list):
+        te2_value, te1_value = in_values
+        if isinstance(te1_value, list):
+            te1_dict = te1[1]
+            if isinstance(te1_dict, dict):
+                te1 = te1_dict.get('EchoTime')
+        if isinstance(te2_value, list):
+            te2_dict = te2[1]
+            if isinstance(te2_dict, dict):
+                te2 = te2_dict.get('EchoTime')
+
+    # For convienience if both are missing we should give one error about them
+    if te1 is None and te2 is None:
+        raise RuntimeError('EchoTime metadata fields not found. '
+                           'Please consult the BIDS specification.')
+    if te1 is None:
+        raise RuntimeError(
+            'EchoTime metadata field not found for phase1. Please consult the BIDS specification.')
+    if te2 is None:
+        raise RuntimeError(
+            'EchoTime metadata field not found for phase2. Please consult the BIDS specification.')
+
+    return abs(float(te2) - float(te1))
```

Comment:
I suspect you want to preserve the sign, so that you can subtract phase1 from phase2, and let the negative delta_te correct the sign if te1 is longer than te2.

Reply:
Sorry for the mess in these, I was mostly wondering if the handling in `init_sdc_wf` looked ok. I'm still filling out the details in the other functions

Reply:
Oh, okay. I'll tag this as `[WIP]`. Just ping us when you want a review. (Or remove the `[WIP]` tag from the title to let us know if you think it's basically ready.)